CONTRIBUTING.rst

```diff
@@ -0,0 +1,177 @@
+
+- Click on the ``compare & pull request`` button to prepare your PR.
+- It is time to communicate your changes: write a title and a description for your PR.
+  People who review your PR are happy to know
+
+  * what feature/fix you propose, and why
+  * how you made it (created a new class than inherits from...)
+  * and anything relevant to your PR (performance tests, images, *etc.*)
+- Press ``Create pull request``.
+  Now you can navigate through your PR, which highlights the changes you made.
+
+Please DO NOT write large Pull Requests, as they are very difficult and time-consuming to review.
+As much as possible, split them into small targeted PRs.
+For example, if find typos in the documentation open a pull request that only fixes typos.
+If you want to fix a bug, make a small pull request that only fixes a bug.
+If you want to implement a large feature, write helper functionality first, test it and submit those as a first pull request.
+If you want to implement a feature and are not too sure how to split it, just open an issue about your plans and ping other WarpX developers on it to chime in.
+
+Even before your work is ready to merge, it can be convenient to create a PR (so you can use Github tools to visualize your changes).
+In this case, please put the ``[WIP]`` tag (for Work-In-Progress) at the beginning of the PR title.
+
+Include a test to your PR
+"""""""""""""""""""""""""
+
+A new feature is great, a **working** new feature is even better!
+Please test your code and add your test to the automated test suite.
+It's the way to protect your work from adventurous developers.
+Instructions are given in the :ref:`testing section <developers-testing>` of our `developer's documentation <https://warpx.readthedocs.io/en/latest/developers/testing.html>`_.
+
+Include documentation to your PR
+""""""""""""""""""""""""""""""""
+
+Now, let users know about your new feature by adding it to the `WarpX documentation <https://warpx.readthedocs.io>`_.
+Our documentation uses `Sphinx <http://www.sphinx-doc.org/en/master/usage/quickstart.html>`_, and it is located in ``Docs/``.
+For instance, if you introduce a new runtime parameter in the input file, you can add it to :ref:`Docs/source/running_cpp/parameters.rst <running-cpp-parameters>`.
+If Sphinx is installed on your computer, you should be able to generate the html documentation with
+
+.. code-block:: sh
+
+   make html
+
+in ``Docs/``. Then open ``Docs/build/html/index.html`` with your favorite web browser and look
+for your changes.
+
+Once your code is ready with documentation and automated test, congratulations!
+You can create the PR (or remove the ``[WIP]`` tag if you already created it).
+Reviewers will interact with you if they have comments/questions.
+
+Style and conventions
+---------------------
+
+- For indentation, WarpX uses four spaces (no tabs)
+- The number of characters per line should be <100
```

Comment:
```suggestion
- The number of characters per line should be <100 (except in documentation files, where each line should be the length of a full sentence, to allow for easy changes/corrections in GitHub)
```

Reply:
Urgh, I had that in here but accidentally removed all changes on the first go before commit xD